test/checkpoint/checkpointer.c

```diff
@@ -261,12 +222,26 @@ real_checkpointer(void)
         }
 
         /* Execute a checkpoint */
-        if ((ret = session->checkpoint(session, checkpoint_config)) != 0)
+        if ((ret = session->checkpoint(
+               session, flush_tier ? flush_tier_config : checkpoint_config)) != 0)
             return (log_print_err("session.checkpoint", ret, 1));
         printf("Finished a checkpoint\n");
         fflush(stdout);
+        if (flush_tier) {
+            /*
+             * FIXME: when we change the API to notify that a flush_tier has completed, we'll need
+             * to set up a general event handler and catch that notification, so we can pass the
+             * flush_tier "cookie" to the test utility function.
+             */
+            testutil_tiered_flush_complete(&g.opts, NULL);
+            flush_tier = false;
+            printf("Finished a flush_tier\n");
 
-        if (!g.running)
+            /* Set up a random delay up to 5000 microseconds for the next flush. */
+            g.opts.tiered_flush_interval_us = __wt_random(&rnd) % 5001;
```

Comment:
This change was somewhat intentional, but not completely well thought out.  Instead of (original) 5000 - 10000 or my first change (0 - 5000) I'm going to make it a random interval between (0- 10000).  I agree that it _shouldn't_ matter, but maybe (an occasional) very quick interval between flush_tier calls will stress something.  And having potential intervals up to 10000 means that we'll get some similar coverage as the original.